examples/blackjack.py

```diff
@@ -0,0 +1,182 @@
+        self.deck.shuffle()
+        print("All shuffled!")
+        print("Dealing...")
+        self.deal()
+        print("\nLet's play!")
+        for player in self.players:
+            print("{}'s turn...".format(player.name))
+            self.play(player)
+        else:
+            print("That's the last turn. Determining the winner...")
+            self.find_winner()
+
+    def play(self, player):
+        """
+        An individual player's turn.
+
+        If the player's cards are an ace and a ten or court card,
+        the player has a blackjack and wins.
+
+        If a player's cards total more than 21, the player loses.
+
+        Otherwise, it takes the sum of their cards and determines whether
+        to hit or stand based on their current score.
+        """
+        while True:
+            points = sum_hand(player.hand)
+            if points < 17:
+                print("   Hit.")
+                self.hit(player)
+            elif points == 21:
+                print("   {} wins!".format(player.name))
+                sys.exit(0) # End if someone wins
+            elif points > 21:
+                print("   Bust!")
+                break
+            else:  # Stand if between 17 and 20 (inclusive)
+                print("   Standing at {} points.".format(str(points)))
+                self.scores[player.name] = points
+                break
+
+    def find_winner(self):
+        """
+        Finds the highest score, then finds which player(s) have that score,
+        and reports them as the winner.
+        """
+        winners = []
+        try:
+            win_score = max(self.scores.values())
+            for key in self.scores.keys():
+                if self.scores[key] == win_score:
+                    winners.append(key)
+                else:
+                    None
+            winstring = " & ".join(winners)
+            print("And the winner is...{}!".format(winstring))
+        except ValueError:
+            print("Whoops! Everybody lost!")
+
+    def deal(self):
+        """
+        Deals two cards to each player.
+        """
+        i = 1
+        while i <= 2:
+            for p in self.players:
+                newcard = self.deck.draw()
+                p.hand.append(newcard)
+                print("Dealt {} the {}.".format(p.name, str(newcard)))
+            i += 1
+
+
+    def hit(self, player):
+        """
+        Adds a card to the player's hand and states which card was drawn.
+        """
+        newcard = self.deck.draw()
+        player.hand.append(newcard)
+        print("   Drew the {}.".format(str(newcard)))
+
+
+def sum_hand(hand: list):
+    """
+    Converts ranks of cards into point values for scoring purposes.
+    'K', 'Q', and 'J' are converted to 10.
+    'A' is converted to 1 (for simplicity), but if the first hand is an ace
+    and a 10-valued card, the player wins with a blackjack.
+    """
+    vals = [card.rank for card in hand]
+    for i in range(0, len(vals)):
```

Comment:
This could be simplified with `for value in vals:`. Using i in iterations over list is considered nonpythonic, it's more suited for languages like C and Java